lib/AlternativeNamesFieldArray/AlternativeNamesFieldArray.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Field } from 'react-final-form';
+
+import {
+  Button,
+  Col,
+  Headline,
+  IconButton,
+  TextField,
+  Row,
+} from '@folio/stripes/components';
+
+import withKiwtFieldArray from '../withKiwtFieldArray';
+
+class AlternativeNamesFieldArray extends React.Component {
+    static propTypes = {
+      items: PropTypes.arrayOf(PropTypes.object),
+      name: PropTypes.string.isRequired,
+      onAddField: PropTypes.func.isRequired,
+      onDeleteField: PropTypes.func.isRequired,
+    }
+
+    static defaultProps = {
+      items: [],
+    }
+
+    renderNames = () => {
+      const { items, name, onDeleteField } = this.props;
+      return items.map((item, index) => (
+        <div
+          key={index}
+          data-test-render-alternative-names
+        >
+          <Row>
+            <Col xs={11}>
+              <Field
```

Comment:
Sorry for not being super clear on the autofocus part. It's important to not _always_ have autofocus on the Field, but we do need to have it focus when a new alternative name is added. 

Eg, note how when you add a new document on a section with DocumentsFieldArray, the Name field is automatically focused instead of focus remaining with the `Add supplementary document` (or similar) button?

We do it by separating out the `DocumentField` parts into their own component, and when that component mounts, [we check whether we should autofocus a particular element](https://github.com/folio-org/stripes-erm-components/blob/5034b0bc4379698ed054feadc0e986ae6c98b280/lib/DocumentsFieldArray/DocumentField.js#L38). In that `if` case, we're checking whether the document object we're editing has an `id`. If it does, it was generated by the backend when that document was initially saved. If it doesn't have an `id` and the component mounted, we can then conclude that the document was added right now by the user pressing an **Add document** button.

So similarly, you'll need to break this component out into an `AlternativeNamesField` and `AlternativeNamesFieldArray` component and use similar `this.inputRef` and `componentDidMount` logic to autofocus the textfield, but _only_ if it's being mounted because the user _just_ added the alternative name, and not if we're loading the Edit Agreement page.